### PR TITLE
Add yamlfmt setup action

### PR DESCRIPTION
**Pull request by puerco**
This commit adds a new action called "setup-yamlfmt" that installs yamlfmt into the actions environment.

Signed-off-by: Adolfo García Veytia (Puerco) <puerco@chainguard.dev>

### CODE REVIEW

**Comment by puerco** [comment]
Mmh I think the linter is wrong here :thinking:

File: setup-yamlfmt/action.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        tip)
+          echo "Installing yamlfmt using go install"
+          go install github.com/google/yamlfmt@main
+          ;;
+        latest)
+          tag=$(curl -L -s -u "username:${{ github.token }}" https://api.github.com/repos/google/yamlfmt/releases/latest | jq -r '.tag_name')
+          ;;
+        *)
+          tag="${{ inputs.version }}"
+        esac
```
Comment (luhring) [inline setup-yamlfmt/action.yaml:36]: I _think_ the `go install <repo>@<git-ref>` syntax handles `main` and `latest` automatically, in addition to tagged versions. Would it make sense to let the `go install` command handle the version parsing here?

**Comment by luhring** [comment]
> Mmh I think [the linter is wrong here](https://github.com/chainguard-dev/actions/actions/runs/3842453584/jobs/6543766083)? 🤔

Yeah, I think you're right. It looks like it's treating this YAML file as a regular GitHub Actions workflow, not as an _action_. 🤔

**Comment by cpanato** [comment]
> Mmh I think [the linter is wrong here](https://github.com/chainguard-dev/actions/actions/runs/3842453584/jobs/6543766083)? 🤔

lets ignore this one

**Review by cpanato — CHANGES_REQUESTED** [review]
rename the action to use `.yml` and not `.yaml`

`setup-yamlfmt/action.yml`

**Comment by cpanato** [comment]
@puerco do still want this?

**Comment by cpanato** [comment]
@puerco gently ping

**Comment by cpanato** [comment]
closing this

**Pull request closed without merging**